src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.dao.FruitStorageDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverterService;
+import core.basesyntax.service.FileReaderService;
+import core.basesyntax.service.FileWriterService;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.impl.CsvFormatReportGenerator;
+import core.basesyntax.service.impl.DataConverterServiceImpl;
+import core.basesyntax.service.impl.FileReaderServiceImpl;
+import core.basesyntax.service.impl.FileWriterServiceImpl;
+import core.basesyntax.service.impl.ShopServiceImpl;
+import core.basesyntax.service.impl.operation.BalanceOperation;
+import core.basesyntax.service.impl.operation.OperationHandler;
+import core.basesyntax.service.impl.operation.PurchaseOperation;
+import core.basesyntax.service.impl.operation.ReturnOperation;
+import core.basesyntax.service.impl.operation.SupplyOperation;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String INPUT_FILE_PATH = "src/main/resources/reportToRead.csv";
+    private static final String OUTPUT_FILE_PATH = "src/main/resources/finalReport.csv";
+    private final FileReaderService fileReader;
+    private final DataConverterService dataConverter;
+    private final OperationStrategy operationStrategy;
+    private final ShopService shopService;
+    private final ReportGenerator reportGenerator;
+    private final FileWriterService fileWriter;
+
+    public Main(FileReaderService fileReader,
+                DataConverterService dataConverter,
+                OperationStrategy operationStrategy,
+                ShopService shopService,
+                ReportGenerator reportGenerator,
+                FileWriterService fileWriter) {
+        this.fileReader = fileReader;
+        this.dataConverter = dataConverter;
+        this.operationStrategy = operationStrategy;
+        this.shopService = shopService;
+        this.reportGenerator = reportGenerator;
+        this.fileWriter = fileWriter;
+    }
```

Comment:
Let's remove it. We don't need to create an instance of Main class, because we have static method main(). 
In general, it's the right way to declare dependencies and initialize it. You will do the same in book-store project, for example
```suggestion
    private final FileReaderService fileReader;
    private final DataConverterService dataConverter;
    private final OperationStrategy operationStrategy;
    private final ShopService shopService;
    private final ReportGenerator reportGenerator;
    private final FileWriterService fileWriter;

    public Main(FileReaderService fileReader,
                DataConverterService dataConverter,
                OperationStrategy operationStrategy,
                ShopService shopService,
                ReportGenerator reportGenerator,
                FileWriterService fileWriter) {
        this.fileReader = fileReader;
        this.dataConverter = dataConverter;
        this.operationStrategy = operationStrategy;
        this.shopService = shopService;
        this.reportGenerator = reportGenerator;
        this.fileWriter = fileWriter;
    }
```

Reply:
changed